src/particle-analysis/EventCharacteristics.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from ParticleClass import Particle
+from Lattice3D import Lattice3D
+
+class EventCharacteristics:
+    """
+    This class computes event-by-event characteristics, e.g., eccentricities
+    or certain densities.
+
+    Parameters
+    ----------
+    particle_data: list, numpy.ndarray
+        List or array containing particle objects for one event.
+
+    Attributes
+    ----------
+    particle_data_: list, numpy.ndarray
+        List or array containing particle objects for one event.
+
+    Methods
+    -------
+    eccentricity_from_particles:
+        Computes the spatial eccentricity from particles.
+    """
+    def __init__(self,particle_data):
+        # check that the input is a list/numpy.ndarray containing Particle objects
+        if not isinstance(particle_data, (list, np.ndarray)):
+            raise TypeError('The input is not a list nor a numpy.ndarray.')
+        for particle in particle_data:
+            if not isinstance(particle, Particle):
+                raise TypeError('At least one element in the input is not a ' +\
+                                'Particle type.')
+        self.particle_data_ = particle_data
+
+    
+    def eccentricity_from_particles(self,harmonic_n):
+        real_eps=0
+        imag_eps=0
+        norm=0
+        for particle in particle_data:
+            x = particle.x
+            y = particle.y
+            rn = x**harmonic_n+y**harmonic_n
```

Comment:
Shouldn't this be `rn = (x**2. + y**2.)**(harmonic_n/2.)`, because of $r=\sqrt{x^2+y^2}$ and there should be an if statement if someone computes $\varepsilon_1$. This is usually computed with a $r^3$ weight instead of $r$.

Reply:
Indeed you are right. I was not aware of the special case for dipole asymmetry.